Sources/Document/DocEvent.swift

```diff
@@ -47,54 +47,60 @@ public protocol DocEvent {
  * the server.
  *
  */
-struct SnapshotEvent: DocEvent {
+public struct SnapshotEvent: DocEvent {
     /**
      * ``DocEventType.snapshot``
      */
-    let type: DocEventType = .snapshot
+    public let type: DocEventType = .snapshot
     /**
      * SnapshotEvent type
      */
-    var value: Data
+    public var value: Data
+}
+
+protocol ChangeEventable: DocEvent {
```

Comment:
How about renaming it to `ChangeEvent` or some other noun as the protocol defines what the implementation should be rather than what it is capable of doing.
Besides, it already inherits from `DocEvent` which is also a noun.